### PR TITLE
chore(vault-csi-provider): update provider to 0.0.7 and dep to 0.0.20

### PR DESCRIPTION
**Pull request by eyenx**
# Description

* Updates our vault-csi-provider to use image tag 0.0.7 and adds default resources, as in the deployment example [changelog](https://github.com/hashicorp/secrets-store-csi-driver-provider-vault/releases/tag/0.0.7)
* Updates the dependency of secrets-store-csi-driver to 0.0.20 [changelog](https://github.com/kubernetes-sigs/secrets-store-csi-driver/releases/tag/v0.0.20)

# Issues

n/a

# Checklist

* [x] I updated the version in Chart.yaml
* [x] I updated applicable README.md files using  `pre-commit run -a`
* [x] I documented any high-level concepts I'm introducing in `docs/`
* [x] CI is currently green and this is ready for review
* [x] I am ready to test changes after they are applied and released

### CODE REVIEW

**Review by pree — APPROVED** [review]
LGTM

File: charts/vault-csi-provider/templates/daemonset.yaml [hunk 1]
```diff
@@ -30,3 +30,5 @@ spec:
         - name: providervol
           hostPath:
             path: {{ .Values.csiProviderPath }}
+      nodeSelector:
+        beta.kubernetes.io/os: linux
```
Comment (tongpu) [inline charts/vault-csi-provider/templates/daemonset.yaml:34]: Is Karbon still using this with a beta prefix? `beta.kubernetes.io/os` is deprecated according to the [docs](https://kubernetes.io/docs/reference/labels-annotations-taints/) and `kubernetes.io/os` should be used instead.
```suggestion
        kubernetes.io/os: linux
```

**Review by tongpu — COMMENTED** [review]
Please double check regarding the `kubernetes.io/os` label.

File: charts/vault-csi-provider/templates/daemonset.yaml (hunk 1 above)
Comment (pree) [reply charts/vault-csi-provider/templates/daemonset.yaml:34]: It's the same on the upstream secrets-store-csi-driver-provider-vault: https://github.com/hashicorp/secrets-store-csi-driver-provider-vault/blob/master/deployment/provider-vault-installer.yaml#L72

Comment (pree) [reply charts/vault-csi-provider/templates/daemonset.yaml:34]: `beta.` was deprecated in 1.14 and the csi-provider needs at least 1.15 - I think we can safely use `kubernetes.io/os`

File: charts/vault-csi-provider/README.md [hunk 2]
```diff
@@ -13,15 +13,15 @@ This chart is maintained by [Adfinis](https://adfinis.com/?pk_campaign=github&pk
...
 
 ## Values
 
 | Key | Type | Default | Description |
 |-----|------|---------|-------------|
 | csiProviderPath | string | `"/etc/kubernetes/secrets-store-csi-providers"` | set the path where the secrets-store-csi-provider gets installed on the node |
-| image | object | `{"pullPolicy":"IfNotPresent","repository":"hashicorp/secrets-store-csi-driver-provider-vault","tag":"0.0.6"}` | specifies the image to use for the secrets-store-csi-driver |
-| resources | object | `{}` | Set the limits and requests on vault csi-driver-provider pod resources |
+| image | object | `{"pullPolicy":"IfNotPresent","repository":"hashicorp/secrets-store-csi-driver-provider-vault","tag":"0.0.7"}` | specifies the image to use for the secrets-store-csi-driver |
+| resources | object | `{"limits":{"cpu":"50m","memory":"100Mi"},"requests":{"cpu":"50m","memory":"100Mi"}}` | Set the limits and requests on vault csi-driver-provider pod resources |
```
Comment (hairmare) [inline charts/vault-csi-provider/README.md:24]: Can you update the doc so they don't contain extremely long JSON strings?

File: charts/vault-csi-provider/templates/daemonset.yaml (hunk 1 above)
Comment (hairmare) [reply charts/vault-csi-provider/templates/daemonset.yaml:34]: In any case (independant of if there is a beta beta prefix or not), this nodeSelector should be confgured via values and not be a hardcoded magic string in the Daemonset.

File: charts/vault-csi-provider/values.yaml [hunk 3]
```diff
@@ -6,11 +6,17 @@
...
 image:
   repository: hashicorp/secrets-store-csi-driver-provider-vault
-  tag: 0.0.6
+  tag: 0.0.7
   pullPolicy: IfNotPresent
 
 # resources -- Set the limits and requests on vault csi-driver-provider pod resources
-resources: {}
+resources:
+  requests:
```
Comment (hairmare) [inline charts/vault-csi-provider/values.yaml:14]: Why deviate from the default of leaving this empty? I can't find a mention of leaving it empty in the best-practices but an empty chart made with `helm create` has this info:
>   We usually recommend not to specify default resources and to leave this as a conscious
   choice for the user. This also increases chances charts run on environments with little
   resources, such as Minikube. If you do want to specify resources, uncomment the following
   lines, adjust them as necessary, and remove the curly braces after 'resources:'.

IMO leaving this as a conscious choise to the user still holds true.

**Review by hairmare — CHANGES_REQUESTED** [review]
There are a few magic strings that should not be hardcoded:

File: charts/vault-csi-provider/templates/daemonset.yaml (hunk 1 above)
Comment (tongpu) [reply charts/vault-csi-provider/templates/daemonset.yaml:34]: Hardcoding a `nodeSelector` is acceptable when we know that the daemonset can't be executed on Windows nodes, but we should probably provide an option to extend the `nodeSelector` or configure our default in `values.yaml`.

Comment (eyenx) [reply charts/vault-csi-provider/templates/daemonset.yaml:34]: what has this to do  with Karbon?

File: charts/vault-csi-provider/values.yaml (hunk 3 above)
Comment (eyenx) [reply charts/vault-csi-provider/values.yaml:14]: It's all taken from the default deployment/daemonset.yaml of the upstream repo. https://github.com/hashicorp/secrets-store-csi-driver-provider-vault/blob/master/deployment/provider-vault-installer.yaml

Comment (eyenx) [reply charts/vault-csi-provider/values.yaml:14]: Would you suggest discarding their upstream input and just leaving this blank?

File: charts/vault-csi-provider/templates/daemonset.yaml (hunk 1 above)
Comment (eyenx) [reply charts/vault-csi-provider/templates/daemonset.yaml]: It's also just taken from upstream here.

Comment (eyenx) [reply charts/vault-csi-provider/templates/daemonset.yaml]: https://github.com/hashicorp/secrets-store-csi-driver-provider-vault/blob/master/deployment/provider-vault-installer.yaml

File: charts/vault-csi-provider/values.yaml (hunk 3 above)
Comment (sinicis) [reply charts/vault-csi-provider/values.yaml:14]: if some resources are required that the pod can work then it makes sense to define the minimal `requests`. `limits` are not needed IMHO

Comment (eyenx) [reply charts/vault-csi-provider/values.yaml]: According to theyr upstream chart which will be released soon, they also set the resources to nil. I will look into that later this month or in April so we can depreacate mantaining `vault-auth` and `vault-csi-provider` ourselfes.

File: charts/vault-csi-provider/README.md (hunk 2 above)
Comment (eyenx) [reply charts/vault-csi-provider/README.md]: Removed.

File: charts/vault-csi-provider/values.yaml (hunk 3 above)
Comment (eyenx) [reply charts/vault-csi-provider/values.yaml]: Removed and set to nil

**Review by sinicis — APPROVED** [review]

**Review by tongpu — APPROVED** [review]

**Pull request merged**